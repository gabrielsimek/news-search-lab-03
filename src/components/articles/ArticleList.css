ul {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
    gap: 50px;
    margin: 20px;
    padding: 0px;
}
