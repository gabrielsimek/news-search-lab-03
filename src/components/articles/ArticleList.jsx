import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleItems = articles.map(article => {
    return (
      <Article
        key={article.title}
        {...article}
      />
    );
  });
  return <ul>{articleItems}</ul>;
};

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      source: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      publishedAt: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,  
    })
  )
};

export default ArticleList;
