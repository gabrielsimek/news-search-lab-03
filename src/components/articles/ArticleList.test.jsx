import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './Article';

describe('ArticleList component', () => {
  afterEach(() => cleanup());
  it('renders ArticleList', () => {
    const { asFragment } = render(<ArticleList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
