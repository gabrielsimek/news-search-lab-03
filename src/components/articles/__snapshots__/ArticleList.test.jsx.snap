// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders ArticleList 1`] = `
<DocumentFragment>
  <li>
    <h1 />
    <img />
    <h2 />
    <p />
    <h2 />
    <h3 />
    <a>
      To Article
    </a>
  </li>
</DocumentFragment>
`;
